pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -274,6 +274,10 @@ func ValidateParameterTypes(ctx context.Context, params []ParamSpec) (errs *apis
 
 // ValidateType checks that the type of a ParamSpec is allowed and its default value matches that type
 func (p ParamSpec) ValidateType() *apis.FieldError {
+	// For resources not call SetDefaults(), empty type should be valid.
+	if p.Type == "" {
```

Comment:
Can you add a unit test in pipeline reconciler that the param type is null, to make sure this wouldn't introduce any new bugs? Thanks